R/bambu-quantify.R

```diff
@@ -95,3 +96,35 @@ bambu.quantDT <- function(readClassDt = readClassDt,
     return(list(theta_est, d_rateOut[1], d_rateOut[2]))
 }
 
+generateReadToTranscriptMap <- function(readClass, annotations){
+    distTable = metadata(readClass[[2]])$distTable
+    readClass = readClass[[1]]
+    if (is.character(readClass)) readClass <- readRDS(file = readClass)
+    if(!is.null(metadata(readClass)$readNames)) { 
+        read_id = metadata(readClass)$readNames}
+    else { read_id = metadata(readClass)$readId}
+
+    #unpack and reverse the read class to read id reletionship
+    readOrder = order(unlist(rowData(readClass)$readIds))
+    lens = lengths(rowData(readClass)$readIds)
+    rcIndex = seq_along(readClass)
+    readToRC = rep(rcIndex, lens)[readOrder]
```

Comment:
you assign readIds based on all reads (` read_id = metadata(readClass)$readId`) but the readClasses are matched based on the reads that are mapped to read classes (`unlist(rowData(readClass)$readIds`). Both objects will have a different length if they are not identical. the matching between readIds from rowData() and readIds form metadata() is missing

Reply:
`read_id= read_id[(match(unlist(rowData(readClass)$readIds)[readOrder], metadata(readClass)$readId))]`

Reply:
something along these lines is missing I think

Reply:
resolved in 7a7db175526f1cdd8c697cc0226e075ff08ed538